src/os/bluestore/BlueStore.cc

```diff
@@ -5266,30 +5269,33 @@ int BlueStore::_setup_block_symlink_or_file(
   if (create)
     flags |= O_CREAT;
   if (epath.length()) {
-    r = ::symlinkat(epath.c_str(), path_fd, name.c_str());
-    if (r < 0) {
-      r = -errno;
-      derr << __func__ << " failed to create " << name << " symlink to "
-           << epath << ": " << cpp_strerror(r) << dendl;
-      return r;
-    }
-
     if (!epath.compare(0, strlen(SPDK_PREFIX), SPDK_PREFIX)) {
       int fd = ::openat(path_fd, epath.c_str(), flags, 0644);
       if (fd < 0) {
 	r = -errno;
 	derr << __func__ << " failed to open " << epath << " file: "
-	     << cpp_strerror(r) << dendl;
+	  << cpp_strerror(r) << dendl;
 	return r;
       }
       string serial_number = epath.substr(strlen(SPDK_PREFIX));
       r = ::write(fd, serial_number.c_str(), serial_number.size());
       assert(r == (int)serial_number.size());
       dout(1) << __func__ << " created " << name << " symlink to "
-              << epath << dendl;
+	<< epath << dendl;
       VOID_TEMP_FAILURE_RETRY(::close(fd));
+    } else if (!create && !fs::exists(epath.c_str())) {
+      return 0;
     }
   }
+
+  r = ::symlinkat(epath.c_str(), path_fd, name.c_str());
```

Comment:
this line should be guarded by `if (epath.length()) {`

that's why quite a few tests failed in http://pulpito.ceph.com/kchai-2018-03-27_02:31:14-rados-wip-kefu-testing-2018-03-26-2040-distro-basic-smithi/